op-node/rollup/driver/state.go

```diff
@@ -443,24 +417,32 @@ func (s *SyncDeriver) onResetEvent(x rollup.ResetEvent) {
 
 // SyncStep performs the sequence of encapsulated syncing steps.
 // Warning: this sequence will be broken apart as outlined in op-node derivers design doc.
-func (s *SyncDeriver) SyncStep() error {
+func (s *SyncDeriver) SyncStep() {
+	s.Log.Debug("Sync process step")
+
+	// Drain errors are safe to ignore:
+	// it only errors on executor context-timeout,
+	// i.e. when we are shutting down and not processing events anymore.
 	if err := s.Drain(); err != nil {
-		return err
+		return
 	}
 
 	s.Emitter.Emit(engine.TryBackupUnsafeReorgEvent{})
 	if err := s.Drain(); err != nil {
-		return err
+		return
 	}
 
 	s.Emitter.Emit(engine.TryUpdateEngineEvent{})
 	if err := s.Drain(); err != nil {
-		return err
+		return
```

Comment:
It feels very wrong to silently ignore all errors here. If a new reason for erroring is added in the future we'll just silently ignore it. Can we return on a context timeout error and emit a `CriticalErrorEvent` on anything else?